webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -0,0 +1,123 @@
+import {EditUserAbstract} from "@mm-modals/edit-user/edit-user.component";
+
+const passwordTester = require('simple-password-tester');
+import {Component} from "@angular/core";
+import {UserSettingsService} from "@mm-services/user-settings.service";
+import {TranslateService} from "@ngx-translate/core";
+import {UpdateUserService} from "@mm-services/update-user.service";
+import {BsModalRef} from "ngx-bootstrap/modal";
+
+const PASSWORD_MINIMUM_LENGTH = 8;
+const PASSWORD_MINIMUM_SCORE = 50;
+
+
+@Component({
+  selector: 'update-password',
+  templateUrl: './update-password.component.html'
+})
+export class UpdatePasswordComponent extends EditUserAbstract {
+
+  editUserModel: {
+    username?,
+    currentPassword?,
+    password?,
+    passwordConfirm?
+  } = {};
+
+  errors: {
+    currentPassword?,
+    password?
+  } = {};
+
+  constructor(
+    bsModalRef: BsModalRef,
+    userSettingsService: UserSettingsService,
+    private translateService: TranslateService,
+    private updateUserService: UpdateUserService,
+  ) {
+    super(bsModalRef, userSettingsService);
+  }
+
+  updatePassword() {
+    this.setProcessing();
+    if (this.validatePasswordFields()) {
+      const updates = { password: this.editUserModel.password };
+      const username = this.editUserModel.username;
+      this.updateUserService.update(username, updates, username, this.editUserModel.currentPassword)
+        .toPromise()
+        .then(() => {
+          this.setFinished();
+          window.location.reload(true);
+        })
+        .catch(err => {
+          if (err.status === -1) { //Offline Status
+            this.translateService.get('online.action.message').toPromise().then(value => {
+              this.errors.currentPassword = value;
+              this.setError(err, value);
+            });
+          } else if (err.status === 401) {
+            this.translateService.get('password.incorrect').toPromise().then(value => {
+              this.errors.currentPassword = value;
+              this.setError(err, value);
+            });
+          } else {
+            this.setError(err, 'Error updating user');
+          }
+        });
+    } else {
+      this.setError();
+    }
+  }
+
+  private validatePasswordFields() {
+    return this.validateRequired('password', 'Password') &&
+      this.validateRequired('currentPassword', 'Current Password') &&
+      this.validatePasswordStrength() &&
+      this.validateConfirmPasswordMatches();
+  }
+
+  private validateRequired(fieldName, fieldDisplayName) {
+    if (!this.editUserModel[fieldName]) {
+      this.translateService
+        .get('field is required', { field: fieldName })
+        .toPromise()
```

Comment:
Does `translateService.get` ever throw? Now I'm asking myself how observables throw :D

Reply:
But the observable is converted into a promise with the `.toPromise()`, so errors should be handled in the catch `.catch(err => {` block below, right? Maybe in practice it never fails, but if somebody in the feature introduce a bug there not detected in the tests, at least will be handled there.